ts/src/binance.ts

```diff
@@ -4392,14 +4392,40 @@ export default class binance extends Exchange {
          * @description fetches information on multiple closed orders made by the user
          * @param {string} symbol unified market symbol of the market orders were made in
          * @param {int|undefined} since the earliest time in ms to fetch orders for
-         * @param {int|undefined} limit the maximum number of  orde structures to retrieve
+         * @param {int|undefined} limit the maximum number of order structures to retrieve
          * @param {object} params extra parameters specific to the binance api endpoint
          * @returns {[object]} a list of [order structures]{@link https://docs.ccxt.com/#/?id=order-structure}
          */
         const orders = await this.fetchOrders (symbol, since, limit, params);
         return this.filterBy (orders, 'status', 'closed');
     }
 
+    async fetchCanceledOrders (symbol: string = undefined, since: Int = undefined, limit: Int = undefined, params = {}) {
+        /**
+         * @method
+         * @name binance#fetchCanceledOrders
+         * @description fetches information on multiple canceled orders made by the user
+         * @see https://binance-docs.github.io/apidocs/spot/en/#all-orders-user_data
+         * @see https://binance-docs.github.io/apidocs/spot/en/#query-margin-account-39-s-all-orders-user_data
+         * @see https://binance-docs.github.io/apidocs/voptions/en/#query-option-order-history-trade
+         * @param {string} symbol unified market symbol of the market the orders were made in
+         * @param {int|undefined} since the earliest time in ms to fetch orders for
+         * @param {int|undefined} limit the maximum number of order structures to retrieve
+         * @param {object} params extra parameters specific to the binance api endpoint
+         * @returns {[object]} a list of [order structures]{@link https://docs.ccxt.com/#/?id=order-structure}
+         */
+        this.checkRequiredSymbol ('fetchCanceledOrders', symbol);
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        if (market['contract'] || market['future']) {
```

Comment:
@carlosmiei @kroitor This should be changed to `if (market['swap'] || market['future'])` because the market['contract'] check prevents it from working with options:
```
binance.fetchCanceledOrders (ETH/USDT:USDT-230429-1900-C)
NotSupported binance fetchCanceledOrders() supports spot, margin and option markets only
---------------------------------------------------
[NotSupported] binance fetchCanceledOrders() supports spot, margin and option markets only

    at fetchCanceledOrders        mnt/c/Users/Dan/Documents/GitHub/ccxt/js/src/binance.js:4489  
    at processTicksAndRejections  node:internal/process/task_queues:96
    at async run                  mnt/c/Users/Dan/Documents/GitHub/ccxt/examples/js/cli.js:295  

binance fetchCanceledOrders() supports spot, margin and option markets only
```

Reply:
will be fixed here shortly: https://github.com/ccxt/ccxt/pull/17766